dags/s2_geomedian_indexing.py

```diff
@@ -0,0 +1,94 @@
+"""
+# Sentinel-2 geomedian product indexing automation
+
+DAG to index Sentinel-2 geomedian backlog data.
+
+"""
+from datetime import datetime, timedelta
+
+from airflow import DAG
+from airflow.kubernetes.secret import Secret
+from airflow.contrib.operators.kubernetes_pod_operator import (
+    KubernetesPodOperator,
+)
+
+DEFAULT_ARGS = {
+    "owner": "Toktam Ebadi",
+    "depends_on_past": False,
+    "start_date": datetime(2020, 6, 14),
+    "email": ["toktam.ebadi@ga.gov.au"],
+    "email_on_failure": False,
+    "email_on_retry": False,
+    "retries": 1,
+    "schedule_interval": "@once",
+    "retry_delay": timedelta(minutes=5),
+    "env_vars": {
+        # TODO: Pass these via templated params in DAG Run
+        "DB_HOSTNAME": "db-writer",
+        "WMS_CONFIG_PATH": "/env/config/ows_cfg.py",
+        "DATACUBE_OWS_CFG": "config.ows_cfg.ows_cfg",
+    },
+    # Lift secrets into environment variables
+    "secrets": [
+        Secret("env", "DB_USERNAME", "odc-writer", "postgres-username"),
+        Secret("env", "DB_PASSWORD", "odc-writer", "postgres-password"),
+        Secret(
+            "env",
+            "AWS_DEFAULT_REGION",
+            "sentinel-2-indexing-user",
+            "AWS_DEFAULT_REGION",
+        ),
+        Secret(
+            "env",
+            "AWS_ACCESS_KEY_ID",
+            "sentinel-2-indexing-user",
+            "AWS_ACCESS_KEY_ID",
+        ),
+        Secret(
+            "env",
+            "AWS_SECRET_ACCESS_KEY",
+            "sentinel-2-indexing-user",
+            "AWS_SECRET_ACCESS_KEY",
+        ),
+        Secret("env", "DB_DATABASE", "odc-writer", "database-name"),
+    ],
+}
+
+INDEXER_IMAGE = "opendatacube/datacube-index:0.0.12"
+
+dag = DAG(
+    "Sentinel-2-geomedian-indexing",
+    doc_md=__doc__,
+    default_args=DEFAULT_ARGS,
+    catchup=False,
+    tags=["k8s", "Sentinel-2-indexing"],
+)
+
+with DAG(
+    "Sentinel-2-geomedian-backlog-indexing",
+    default_args=DEFAULT_ARGS,
+    schedule_interval=DEFAULT_ARGS["schedule_interval"],
+    tags=["Sentinel-2", "geomedian-indexing"],
+    catchup=False,
+) as dag:
+
+    # This needs to be updated in the future in case more zones have been added
+    x = range(153, 246)
```

Comment:
I'm a bit concerned about this, how will this be updated when content of db changes?

Reply:
It won't.  It's very specific to this case.  May be I need to allow the user to specify this range as input parameters to the DAG.

Reply:
Will this address your concern?

Reply:
this needs to be 246+1, range does not include last value and bucket has x246 in it right now